modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/GetObjectDefinitionsRelationsMVCResourceCommand.java

```diff
@@ -0,0 +1,114 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.web.internal.object.definitions.portlet.action;
+
+import com.liferay.object.constants.ObjectPortletKeys;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectRelationship;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectRelationshipLocalService;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Carolina Barbosa
+ */
+@Component(
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + ObjectPortletKeys.OBJECT_DEFINITIONS,
+		"mvc.command.name=/object_definitions/get_object_definitions_relations"
+	},
+	service = MVCResourceCommand.class
+)
+public class GetObjectDefinitionsRelationsMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		JSONArray objectDefinitionsJSONArray = _jsonFactory.createJSONArray();
+
+		List<ObjectDefinition> objectDefinitions =
+			_objectDefinitionLocalService.getObjectDefinitions(
+				_portal.getCompanyId(resourceRequest), true, false,
+				WorkflowConstants.STATUS_APPROVED);
+
+		List<ObjectRelationship> objectRelationships =
+			_objectRelationshipLocalService.getObjectRelationships(
+				ParamUtil.getLong(resourceRequest, "objectDefinitionId"));
+
+		for (ObjectDefinition objectDefinition : objectDefinitions) {
+			objectDefinitionsJSONArray.put(
+				JSONUtil.put(
+					"id", objectDefinition.getObjectDefinitionId()
+				).put(
+					"label",
+					objectDefinition.getLabel(
+						_portal.getLocale(resourceRequest))
+				).put(
+					"related",
+					() -> {
+						if (ListUtil.exists(
```

Comment:
Isn't there a way to get just the other definitions that are related to this specific object definition?

Reply:
I need to send all custom objects (which are published and active), no matter if they are related or not. However, I need to specify whether the object is in a relationship with the object responsible for executing the action (if it's a one-to-many relationship, we should only consider the `n` side). This information will be used by the FE. That's why I decided to get all objects right away and for each one check if it's part of a relationship or not.

Also, I couldn't find an existing way to get only the related objects.

Reply:
It makes sense now. Thanks for explaining the logic.